src/writer/direntPool.h

```diff
@@ -65,22 +73,22 @@ namespace zim
         }
 
         Dirent* getClassicDirent(NS ns, const std::string& path, const std::string& title, uint16_t mimetype) {
-          if (direntIndex == 0xFFFF) {
-            allocate_new_pool();
-          }
-          auto dirent = pools.back() + direntIndex++;
+          auto dirent = getDirentSlot();
           new (dirent) Dirent(ns, path, title, mimetype);
           return dirent;
         }
 
         Dirent* getRedirectDirent(NS ns, const std::string& path, const std::string& title, NS targetNs, const std::string& targetPath) {
-          if (direntIndex == 0xFFFF) {
-            allocate_new_pool();
-          }
-          auto dirent = pools.back() + direntIndex++;
+          auto dirent = getDirentSlot();
           new (dirent) Dirent(ns, path, title, targetNs, targetPath);
           return dirent;
         }
+
+        Dirent* getCloneDirent(const std::string& path, const std::string& title, const Dirent& target) {
```

Comment:
Did this one escape the `Clone` → `Alias` renaming or it is intentional?

Reply:
It was, as at this level, we are creating a Clone.
The difference in behavior (indexing) is done as a higher level.

But keeping the "clone" semantic only here will mostly disturb us.